resources/trigger_aws_test_run.py

```diff
@@ -0,0 +1,205 @@
+"""
+
+import logging
+
+import boto3
+import requests
+
+
+REGION = 'us-west-2'
+PROJECT_NAME = 'edx-app-test'
+PROJECT_ARN = ('arn:aws:devicefarm:us-west-2:372153017832:project:f5eed21b-e123-4867-9fed-aa50c746b813')
+DEVICE_POOL_NAME = 'edx_devices_pool'  # this is a default pool
+DEVICE_POOL_ARN = ('arn:aws:devicefarm:us-west-2:372153017832:devicepool:f5eed21b-e123-4867-9fed-aa50c746b813/ed3586d1-1ff6-4e1d-8220-405b8cd50ba4')
+ANDROID_APP_UPLOAD_TYPE = 'ANDROID_APP'
+PACKAGE_UPLOAD_TYPE = 'APPIUM_PYTHON_TEST_PACKAGE'
+TEST_SPECS_FILE = 'edx.yml'
+TEST_SPECS_ARN = ('arn:aws:devicefarm:us-west-2:372153017832:upload:f5eed21b-e123-4867-9fed-aa50c746b813/2d4859bf-983a-4e8a-b117-4b276d3ae211')
+RUN_TYPE = 'APPIUM_PYTHON'
+RUN_NAME = 'edX test run'
+
+# TODO will be replaced with generated apk path on jenkins
+AUT_NAME = 'edx-2.19.1.apk'
+# TODO will be replaced with generated project path
+PACKAGE_NAME = 'test_bundle.zip'
+
+device_farm = boto3.client('devicefarm', region_name=REGION)
+logger = logging.getLogger(__name__)
+
+
+def aws_job():
+    """
+    aws job to manage test run
+    """
+
+    project_arn = get_project_arn(PROJECT_NAME)
+    print('Project Arn: {}'.format(project_arn))
+    assert project_arn == PROJECT_ARN
+
+    aut_arn = make_upload(project_arn, ANDROID_APP_UPLOAD_TYPE, AUT_NAME)
+    print('{} uploaded successfully'.format(AUT_NAME))
+
+    package_arn = make_upload(project_arn, PACKAGE_UPLOAD_TYPE, PACKAGE_NAME)
+    print('{} uploaded successfully'.format(PACKAGE_NAME))
+
+    device_pool_arn = get_device_pool(project_arn, DEVICE_POOL_NAME)
+    assert device_pool_arn == DEVICE_POOL_ARN
+
+    test_run_arn = schedule_run(
+        project_arn=project_arn,
+        name=RUN_NAME,
+        device_pool_arn=device_pool_arn,
+        app_arn=aut_arn,
+        test_package_arn=package_arn,
+        test_specs_arn=TEST_SPECS_ARN)
+    print('test run Arn: {}'.format(test_run_arn))
+
+    get_test_run(test_run_arn)
+
+
+def get_project_arn(project_name):
+    """
+    get project arn
+
+    Arguments:
+            project_arn (str): project arn
+
+    Return:
+            str: project arn value
+
+    """
+
+    for project in device_farm.list_projects()['projects']:
+        if project['name'] == project_name:
+            return project['arn']
+    raise KeyError('Could not find project %r' % project_name)
+
+
+def make_upload(project_arn, upload_type, target_file_name):
+    """
+    intiate upload and return arn
+
+    Arguments:
+            project_arn (str): project arn
+            upload_type (str): upload type
+            target_file_name (str): file name to upload
+
+    Return:
+            str: uploaded arn
+
+    """
+
+    print('Starting upload {} as {}'.format(target_file_name, upload_type))
+    upload_response = device_farm.create_upload(
+        projectArn=project_arn,
+        name=target_file_name,
+        type=upload_type)
+
+    uploadArn = upload_response['upload']['arn']
```

Comment:
nit: Can you use this `naming_convention` instead of camel case?

Reply:
done